spec/finish_unfinished_persons_spec.rb

```diff
@@ -0,0 +1,70 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe FinishUnfinishedPersons do
+  describe '.compute_semi_id' do
+    let(:competition_year) { 2023 }
+    let(:available_per_semi) { {} }
+
+    before do
+      allow(Person).to receive_messages(
+        where: Person,
+        order: Person,
+        pick: nil,
+      )
+    end
+
+    context 'when a unique semi-ID can be generated' do
+      let(:person_name) { 'John Doe' }
+
+      it 'returns the correct semi-ID and updates available_per_semi' do
+        semi_id, updated_available = described_class.compute_semi_id(competition_year, person_name, available_per_semi)
+
+        expect(semi_id).to start_with('2023')
+        expect(updated_available).to have_key(semi_id)
+        expect(updated_available[semi_id]).to eq(98) # 99 - 1
+      end
+    end
+
+    context 'when IDs are already taken' do
+      let(:person_name) { 'John Doe' }
+
+      before do
+        allow(Person).to receive(:pick).and_return('2023DOEJ01')
```

Comment:
If you want the `Person` entity (which just represents the underlying database table) to return a certain value, then insert that value into the database using `FactoryBot`.